### ISSUE TITLE
Proposal to add a platform-mountable clutter sensor/simulator

### ISSUE THREAD

**Issue by ekhunter123** [dstl/Stone-Soup#478]
**Problem:**
None of the current Sensors are able to add clutter to the detections. For instance in the [UAV demonstration](https://stonesoup.readthedocs.io/en/latest/auto_demos/UAV_tutorial.html) where the groundtruth is given and detections are simulated, it would be nice to add clutter to help test the filters in more a complex simulation.

**Solution:**
I propose the addition of a ClutterSensor class as a subclass of the Sensor class (which is itself a subclass of PlatformMountable). This type of Sensor could be mounted onto a Platform just like the other sensors. It would simulate clutter across the measurement space. The default distribution for the clutter will be uniform, but the user could set it to be others (ie a normal distribution). 

**Expected Usage** (following the example of adding clutter to the UAV demonstration):
<pre><code>sensor = RadarElevationBearingRange(...)    
clutter_sensor = ClutterSensor(
        position_mapping=[0, 2, 4],
        ndim_state=6,
        clutter_rate=2.0,
        measurement_space=[[x_min, x_max], [y_min, y_max]], #where the variables are pre-set
        distribution='uniform')
platform = FixedPlatform(..., sensors=[sensor, clutter_sensor])  
detector = PlatformDetectionSimulator(ground_truth_reader, [platform])  </code></pre>

I would love anyone's thoughts or comments on this! Is there something like this already in the works? If not, I'm happy to work on it. Also let me know if there is an easy work-around already implemented that I am missing!

**Comment by DaveKirkland** [comment]
It is somewhat convenient to add clutter directly via the platform. For simple clutter that is probably ok. Trying to simulate clutter in other ways would probably require Clutter as a separate class that generates detections and needs information on the sensor (position, beamwidth etc). 

Consider a maritime scenario with sea surface clutter and 2 spatially separated radar sensing platforms. The clutter that appears from the same area of the sea surface may be different due to the different aspect angles of the platforms/sensors.

I suspect adding a clutter target class may have additional effects on the StoneSoup design. Just some quick thoughts. I'd be interested in what others have to say.

**Comment by gawebb-dstl** [comment]
I’ve only seen clutter modelling in Stone-Soup in the [Data Association Tutorial](https://github.com/dstl/Stone-Soup/blob/main/docs/tutorials/05_DataAssociation-Clutter.py):
```truth_x = state.state_vector[0]
truth_y = state.state_vector[2]
for _ in range(np.random.randint(10)):
        x = uniform.rvs(truth_x - 10, 20)
        y = uniform.rvs(truth_y - 10, 20)
        measurement_set.add(Clutter(np.array([[x], [y]]), timestamp=state.timestamp, measurement_model=measurement_model))```

I agree with Dave, either the method above or Emily's method is good for simple clutter. However a different method may be needed for more complex clutter models.

You could introduce clutter as additional targets to measure, with a location that changes completely on each timestep. That would mean you'd only ever get clutter in your sensor's field of view. To model Dave's martime scenario above, you'd have to introduce variable angular dependent RCS (or something similar) to the clutter targets to make their detection dependent on the position of the sensor.

**Comment by ekhunter123** [comment]
I like your idea of introducing the clutter as additional targets! I'll keep playing around with a simple clutter model to start.

**Comment by sdhiscocks** [comment]
Resolved with 9499a69
